src/components/plotly-modal/index.js

```diff
@@ -19,39 +19,46 @@ const PlotlyModal = ({ metadata, onToggle, visible }) => {
     return null;
   }
   return (
-    <div className="pipeline-plotly-modal">
-      <div className="pipeline-plot-modal__top">
-        <button
-          className="pipeline-plot-modal__back"
-          onClick={onCollapsePlotClick}
-        >
-          <BackIcon className="pipeline-plot-modal__back-icon"></BackIcon>
-          <span className="pipeline-plot-modal__back-text">Back</span>
-        </button>
-        <div className="pipeline-plot-modal__header">
-          <NodeIcon className="pipeline-plot-modal__icon" icon={nodeTypeIcon} />
-          <span className="pipeline-plot-modal__title">
-            {metadata.node.name}
-          </span>
+    <>
+      {metadata && metadata.plot && (
```

Comment:
you can potentially move this check to line 18 and short circuit there. IMO it's cleaner that way.

```
if (!visible.plotModal || metadata?.plot) {
    return null
}
```

Reply:
Optional chaining for the win. Forgot this also works in vanilla JS, not just TS.

@rashidakanchwala I agree with Lim here. This is cleaner.